sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java

```diff
@@ -49,6 +62,55 @@ public EditActionSteps(WebDriverHelpers webDriverHelpers, BaseSteps baseSteps) {
           Action collectedAction = collectActionData();
           ComparisonHelper.compareEqualEntities(action, collectedAction);
         });
+
+    When(
+        "I click on ([^\"]*) button from NEW DOCUMENT in Event Action tab",
+        (String buttonName) -> {
+          webDriverHelpers.clickOnWebElementBySelector(NEW_DOCUMENT_BUTTON);
+          webDriverHelpers.clickWebElementByText(START_DATA_IMPORT_BUTTON, buttonName);
+        });
+    When(
+        "I upload ([^\"]*) file to the Event Action",
+        (String fileType) -> {
+          webDriverHelpers.sendFile(FILE_PICKER, userDirPath + "/uploads/" + fileType);
+          TimeUnit.SECONDS.sleep(2); // wait for upload file
+        });
+    When(
+        "I check if ([^\"]*) file is available in Event Action documents",
+        (String fileType) -> {
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
+              By.xpath(String.format(CASE_UPLOADED_TEST_FILE, fileType)), 5);
+          TimeUnit.SECONDS.sleep(2);
+        });
+    When(
+        "I check if ([^\"]*) file is downloaded correctly from Event Action tab",
+        (String filename) -> {
+          String file = userDirPath + "/downloads/" + filename;
+          Path path = Paths.get(file);
+          softly.assertTrue(Files.exists(path));
```

Comment:
assert message missing

Reply:
assertHelpers.assertWithPoll(
              () ->
                  Assert.assertTrue(
                      Files.exists(path),
                      dictionaryName + " wasn't downloaded: " + path.toAbsolutePath()),
              30); 

And you can use this approach to wait after file to be downloaded and check it